Fix -Wimplicit-function-declaration

On modern macOS implicit function declarations are now a compile error,
because Apple's new M1 chips use a different calling convention for
variadic functions (and the compiler thus needs to know the function
declaration to figure out which calling convension to use for the
invocation).

Include signal.h for kill(2) and declare set_progname() in io.h to fix
two implicit declaration warnings.

Upstream-Status: Submitted [email to maintainer]
--- ./io.h.orig	2021-02-12 19:51:53.000000000 +0100
+++ ./io.h	2021-02-12 19:51:34.000000000 +0100
@@ -9,5 +9,6 @@
 int     edup            (int oldfd);
 int     edup2           (int oldfd, int newfd);
 FILE*   efdopen         (int fd, const char *mode);
+void    set_progname    (const char *name);
 
 #endif
--- ./ttyrec.c.orig	2021-02-12 19:49:11.000000000 +0100
+++ ./ttyrec.c	2021-02-12 19:49:33.000000000 +0100
@@ -49,6 +49,7 @@
 #include <sys/time.h>
 #include <sys/file.h>
 #include <sys/signal.h>
+#include <signal.h>
 #include <stdio.h>
 #include <time.h>
 #include <unistd.h>
